src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java

```diff
@@ -133,6 +134,27 @@ public ResponseEntity<Object> updateOverseasEntity(Transaction transaction,
         return ResponseEntity.ok().build();
     }
 
+    public ResponseEntity<Object> getSavedOverseasEntity(Transaction transaction,
+                                                         String submissionId,
+                                                         String requestId) throws SubmissionNotFoundException {
+        ApiLogger.debugContext(requestId, "Called getOverseasEntity(...)");
+
+        final String submissionUri = getSubmissionUri(transaction.getId(), submissionId);
+
+        if (!transactionUtils.isTransactionLinkedToOverseasEntitySubmission(transaction, submissionUri)) {
+            return ResponseEntity.badRequest().body(String.format(
```

Comment:
i think we should log an error here before we return the badRequest so that if it does happen we can tell what has happened when we look in the logs. So use an errorContext with tranasction_id and submission_id.
we should probably add the logging in to the update method as well (where it also returns badRequest for the same check)

Reply:
actually, the service shouldn't return a response entity, that should be done by the controller, so it would be better here to throw a SubmissionNotLinkedToTransactionException and catch and log that in the controller before returning the response entity from the controller

Reply:
I'm not sure why this wasn't raised when updateOverseasEntity was written. I have based this on whats's there I did think this myself but as it had been previously approved. I just went ahead with it as we had no specific exception for this, which I assumed was by design.

Reply:
probably just something that was overlooked, better that we spot it now, seems a lot better and more consistent with the new approach, thnaks.